### ISSUE TITLE
[Feature Request] Gitlab oAuth

### ISSUE THREAD

**Issue by cfarence** [digitalfondue/lavagna#8]
It would be nice if you could use a self hosted Gitlab as a login providor. This would go well with #7, since gitlabs web hooks are similar to Github's it should be easy to support both.

**Comment by syjer** [comment]
Yes, agree, in the overhaul of the authentication system, I'll integrate a more flexible solution for the oauth part :)
